tests/Doctrine/Tests/ORM/Mapping/AbstractMappingDriverTest.php

```diff
@@ -359,6 +360,62 @@ public function testBooleanValuesForOptionIsSetCorrectly()
         return $class;
     }
 
+    public function testOwningSideResolution() : void
```

Comment:
Please split this into multiple tests, each `//` can be a separate method. Things would be much simpler to understand.